Source/JavaScriptCore/runtime/MathObject.cpp

```diff
@@ -450,7 +450,10 @@ JSC_DEFINE_HOST_FUNCTION(mathProtoFuncSumPrecise, (JSGlobalObject* globalObject,
     if (iterable.isUndefinedOrNull())
         return throwVMTypeError(globalObject, scope, "Math.sumPrecise requires first argument not be null or undefined"_s);
 
-    PreciseSum sum;
+    uint64_t length = toLength(globalObject, iterable.toObject(globalObject));
+    RETURN_IF_EXCEPTION(scope, encodedJSValue());
```

Comment:
`toLength`, `toObject` etc. are side-effectful. Thus, you cannot do that here. (if an iterable is not array etc., then it performs `get-by-id` for `"length"`, and it is not defined in the spec, and user can observe this behavior by setting a getter for this field).

Reply:
Does that mean there's no way to get the `length` info for the iterator before `forEachInIterable`?

Reply:
Ended up using a template instead of `union` and `enum` because `switch` or `if` statement for `enum` branching makes the code slow.
Additionally, use `XsumLarge` only if `iterable` is an array type, otherwise fall back to `XsumSmall` as before